### DISCUSSION TITLE
What is the purpose of the directories "GraphRoot"/{mounts,tmp} like /var/lib/containers/storage/{mounts,tmp}?

### DISCUSSION THREAD

**Discussion by ttsuuubasa** [containers/podman#16195] [Q&A] [answered]
I'm not sure what is for the directories "GraphRoot"/{mounts,tmp} like /var/lib/containers/storage/{mounts,tmp}.

These directories are created at storage.GetStore() function.
https://github.com/containers/podman/blob/main/vendor/github.com/containers/storage/store.go#L697

However, any commands I tried create nothing under these directories.

I tried running container with bind mount and volume mount like following.
I used podman version 3.3.1.

[bind mount]
```
# ls -l
total 0
drwxr-xr-x. 2 root root 23 Oct 13 17:57 test

# podman run --name mounts1 -d -v ./test:/root/test:Z ubi8/ubi /sbin/init
e29ed5f41e0e13718c53a998fca62e348b65bc6109262478d4e38186f8ac636f

# ls /var/lib/containers/storage/mounts/
#  // nothing created
```

[volume mount]
```
# podman volume create testVol
testVol

# podman run --name mounts2 -d -v testVol:/root/test:Z ubi8/ubi /sbin/init
dac94b7bfb8a9f787d62294d9782ff07e4d1dba323633c0552148f8d01cd7162

# ls /var/lib/containers/storage/mounts/
#  // nothing created
```

What is the purpose of these directories?

**Comment by mheon** [comment]
@nalind Looks like these are used by c/storage, not Libpod

**Comment by nalind** [comment, marked as answer]
They're leftovers from early versions, when we thought we'd end up needing them for something.

**Comment by ttsuuubasa** [comment]
Thank you for your answer.

In that case you said, I think these directories are not needed even now.
If these directories are not planned to use in the future, how about making these directories not create?

**Comment by ttsuuubasa** [comment]
@nalind Could you please comment to the post above?

**Comment by nalind** [comment]
For whatever reason, github didn't create a link to https://github.com/containers/storage/pull/1398, which did just that.

**Comment by ttsuuubasa** [comment]
OK, I think that's fine. I will close this discussion.

A conclusion of this discussion.
\---
"GraphRoot"/{mounts,tmp} subdirectories are no longer created by [containers/storage#1398](https://github.com/containers/storage/pull/1398)
\---
